# Chapter 17 - Exercises

## 17.1

### Answers

* a) No - more than 2 outcomes
  - trial: roll of 1 die
  - n = 50
  [X] outcomes: 1, 2, 3, 4, 5, 6
  - independent: yes
* b) Yes
  - trial: randomly selected person has type A blood
  - n = 20
  - p = 0.43
  - independent: yes
* c) No - trials aren't independent
  - trial: draw of 1 card
  [X] p = 1/4, 12/51, 11/50, 10/49
  [X] independent: No
* d) No - sample sizer larger than 10% - can't treat as independent
  - trial: poll single, randomly selected potential voter
  - n = 500
  - p = ?
  [X] independent: no -> 500 >= 10% of 3k
* e) Yes
  - trial: check whether package is sealed
  - n = 24
  - p = 0.10
  - independent: yes

## 17.3

### Answers

* a) select random number between 1 and 10; 1-2 = success, rest = failure; tally number of draws until success
* b) see below
* c) see below
* d) see below
* e) fairly close to the expected probabilities - weighted a little heavier towards the lower number of draws

In [26]:
import random
from collections import defaultdict

p = 0.2
def draw_until(p):
    counter = 1
    while(p < random.random()):
        counter += 1
    return counter

def trial(n, p):
    counts = defaultdict(int)
    for _ in range(n):
        counts[draw_until(p)] += 1
    return counts

In [27]:
# items b) and c)
t = trial(30, p)
for (k,v) in t.items():
    print("{0}: {1} - {2:%}".format(k, v, v / 30))

1: 5 - 16.666667%
2: 4 - 13.333333%
3: 1 - 3.333333%
20: 1 - 3.333333%
5: 1 - 3.333333%
6: 4 - 13.333333%
7: 5 - 16.666667%
8: 2 - 6.666667%
9: 1 - 3.333333%
10: 2 - 6.666667%
4: 4 - 13.333333%


In [28]:
# item d)
def prob(p, draw):
    return (1 - p) ** (draw - 1) * (p)
    
probs = [(draw, prob(0.2, draw)) for draw in range(1, 11)]
probs

[(1, 0.2),
 (2, 0.16000000000000003),
 (3, 0.12800000000000003),
 (4, 0.10240000000000003),
 (5, 0.08192000000000002),
 (6, 0.06553600000000002),
 (7, 0.052428800000000025),
 (8, 0.041943040000000015),
 (9, 0.033554432000000016),
 (10, 0.026843545600000015)]

## 17.5

### Answers

* a) draw 5 random numbers between 1 and 10; count the number of draws with a value <= 2
* b) see below
* c) see below
* d) see below
* e) similar, but varies slightly from true values

In [29]:
def draw_n(p, n):
    draws = [random.random() for _ in range(n)]
    return sum([1 for draw in draws if draw <= p])

In [30]:
def trial(count, p, n):
    counts = defaultdict(int)
    for _ in range(count):
        counts[draw_n(p, n)] += 1
    return counts

In [36]:
# result for b)
result = trial(30, 0.2, 5)
result

defaultdict(int, {0: 12, 1: 10, 2: 4, 3: 4})

In [37]:
# result for c)
for (k, v) in result.items():
    print("{0} - {1} {2:%}".format(k, v, v / 30))

0 - 12 40.000000%
1 - 10 33.333333%
2 - 4 13.333333%
3 - 4 13.333333%


In [45]:
# result for d)
from scipy.special import binom

def prob(n, p, i):
    return binom(n, i) * p ** i * (1 - p) ** (n - i)
def binom_probs(n, p):
    return [(i, prob(n, p, i)) for i in range(n + 1)]
        
binom_probs(5, 0.2)

[(0, 0.32768000000000008),
 (1, 0.40960000000000008),
 (2, 0.20480000000000009),
 (3, 0.051200000000000023),
 (4, 0.006400000000000002),
 (5, 0.00032000000000000008)]

## 17.7

### Answers

* Not necessarily independent - a delayed flight might cause other flights to be delayed.  Or, the underlying cause for delays might be common across multiple flights.

## 17.9

### Answers

* a) 8.19%
* b) 0.64%
* c) 99.2%

In [48]:
.8 ** 4 * .2
.2 ** 3 * .8
.8 + .16 + .032

0.9920000000000001

## 17.11

### Answers

* E(X) for Geometric model = 1/p; p = 0.2 => E(X) = 5

## 17.13

### Answers

* p = 0.05
* E(X) = 1/p = 1/0.05 => 20

## 17.15

### Answers

* see below

In [49]:
# a
p = 0.04
exp_val = 1 / p
exp_val

25.0

In [50]:
# b
# 1 - P(no AB Donors in first 5)
1 - (0.96 ** 5)

0.18462730240000014

In [52]:
# c
p = 0.04
sum([(1 - p) ** i * p for i in range(6)])

0.21724221030399998

In [54]:
# d
(1 - p) ** 9 #* p

0.69253399582448